docs/docs/guides/operate/managing-concurrency.md

```diff
@@ -107,18 +101,416 @@ concurrency:
+# values.yaml for Dagster Helm chart
+dagster:
+  instance:
+    config:
+      # Concurrency settings for Kubernetes deployment
+      concurrency:
+        runs:
+          max_concurrent_runs: 15
+          tag_concurrency_limits:
+            - key: "kubernetes/namespace"
+              value: "data-pipelines"
+              limit: 10
+            - key: "resource_class"
+              value: "memory_intensive"
+              limit: 3
+            - key: "dagster/backfill"
+              limit: 5
+        
+        pools:
+          granularity: 'op'
+          default_limit: 8
+
+      # Run queue for K8s
+      run_queue:
+        max_concurrent_runs: 15
+        tag_concurrency_limits:
+          - key: "compute_type"
+            value: "gpu"
+            limit: 2
+
+# K8s run launcher configuration
+dagsterUserDeployments:
+  enabled: true
+  deployments:
+    - name: "data-pipeline"
+      image:
+        repository: "my-registry/dagster-user-code"
+        tag: "latest"
+      dagsterApiGrpcArgs:
+        - "--port"
+        - "4000"
+      port: 4000
+      env:
+        DAGSTER_POSTGRES_USER: ${POSTGRES_USER}
+        DAGSTER_POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
+        DAGSTER_POSTGRES_HOST: ${POSTGRES_HOST}
+        DAGSTER_POSTGRES_DB: ${POSTGRES_DB}
+```
+
+### Example 3: Docker Compose Deployment
+
+For Docker Compose deployments, configure concurrency in the shared `dagster.yaml`:
+
+```yaml
+# docker-compose.yml
+version: '3.8'
+services:
+  dagster_webserver:
+    image: my-dagster-image
+    environment:
+      DAGSTER_POSTGRES_USER: dagster
+      DAGSTER_POSTGRES_PASSWORD: dagster
+      DAGSTER_POSTGRES_HOST: postgres
+      DAGSTER_POSTGRES_DB: dagster
+    volumes:
+      - ./dagster.yaml:/opt/dagster/dagster_home/dagster.yaml
+    
+  dagster_daemon:
+    image: my-dagster-image
+    environment:
+      DAGSTER_POSTGRES_USER: dagster
+      DAGSTER_POSTGRES_PASSWORD: dagster
+      DAGSTER_POSTGRES_HOST: postgres
+      DAGSTER_POSTGRES_DB: dagster
+    volumes:
+      - ./dagster.yaml:/opt/dagster/dagster_home/dagster.yaml
+    command: ["dagster-daemon", "run"]
+```
+
+```yaml
+# dagster.yaml for Docker Compose
+storage:
+  postgres:
+    postgres_db:
+      username: ${DAGSTER_POSTGRES_USER}
+      password: ${DAGSTER_POSTGRES_PASSWORD}
+      hostname: ${DAGSTER_POSTGRES_HOST}
+      db_name: ${DAGSTER_POSTGRES_DB}
+      port: 5432
+
+run_launcher:
+  module: dagster_docker
+  class: DockerRunLauncher
+  config:
+    image: "my-dagster-image"
+    network: "dagster_network"
+
+concurrency:
+  runs:
+    max_concurrent_runs: 5
+    tag_concurrency_limits:
+      - key: "docker/network"
+        value: "dagster_network"
+        limit: 3
+      - key: "memory_usage"
+        value: "high"
+        limit: 2
+  
+  pools:
+    default_limit: 3
+```
+
+## Advanced Concurrency Patterns
+
+### Pattern 1: Environment-Based Concurrency
+
+Configure different concurrency limits based on environment:
+
+```python
```

Comment:
Similar to the above recommendations -- I'd put this in the `docs_snippets` folder and reference with `<CodeExample>`